library/ziti.c

```diff
@@ -2137,28 +2137,38 @@ static void api_session_cb(ziti_api_session *api_sess, const ziti_error *err, vo
             goto done;
         }
 
-        struct tm exp;
-        ztx->id_creds.cert->get_expiration(ztx->id_creds.cert, &exp);
-        time_t now = time(0);
-        time_t exptime = mktime(&exp);
-
-        bool renew = exptime - now < ztx->opts.cert_extension_window * ONE_DAY;
-        if (!renew) {
+        if (!api_sess->is_cert_extendable) {
+            ZTX_LOG(DEBUG, "identity certificate is not renewable");
```

Comment:
this feels like a warn to me

Reply:
primary cause of non-renewable certs is 3rd party CA identities. we don't want that noise at WARN level

Reply:
seems like the sorta thing we could test for but i get ya. thx